src/Components/InstanceTypesSelect/index.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { Spinner, FormSelect, FormSelectOption } from '@patternfly/react-core';
+import { useQuery } from 'react-query';
+import { TYPES_QUERY_KEY } from '../../API/queryKeys';
+import { fetchInstanceTypesList } from '../../API';
+import { useGlobalState } from '../Common/GlobalState';
+
+const InstanceTypesSelect = () => {
+  const [globalState, setGlobalState] = useGlobalState();
+  const {
+    isLoading,
+    error,
+    data: instanceTypes,
+  } = useQuery(
+    TYPES_QUERY_KEY,
+    () => fetchInstanceTypesList(globalState.chosenSource),
```

Comment:
- [ ] could you create this function at the top level?  this creates a new function on every re-render

```js
const fetchFunction = () => fetchInstanceTypesList(globalState.chosenSource);
const {
     isLoading,
     error,
     data: instanceTypes,
   } = useQuery(
     TYPES_QUERY_KEY,
     fetchFunction,
   { enabled: !!globalState.chosenSource }